app/src/main/java/org/mozilla/fenix/home/recentbookmarks/view/RecentBookmarksHeaderViewHolder.kt

```diff
@@ -59,7 +59,7 @@ class RecentBookmarksHeaderViewHolder(
                 },
             )
 
-            Spacer(Modifier.height(16.dp))
+            Spacer(Modifier.height(4.dp))
```

Comment:
echoing @gabrielluong's concerns here, the Spacer changes in the viewholders should not be necessary with the change to `TextButton` from `BasicText`

Reply:
Gotcha gotcha. I'd ask we UX check this then to ensure padding amount is what they're expecting.

Reply:
Just want to make sure I am understanding the difference between the screenshots here. If my assumption is correct, the 2 screenshots would be still using the new composable with various spacer changes. However, there is no screenshot of what the current nightly looks like for comparison.

Reply:
I added screenshot for comparison in the PR description, however I didn't provide one to be compared with the spacer changes removed. Here it is:

| Current headers and spacing | Updated header with spacing | Updated header without spacing |
| --- | --- | --- |
| ![b4](https://user-images.githubusercontent.com/35462038/195355330-743a5fe4-5506-49af-939d-5128a4ef13a1.png) | ![after_spacing](https://user-images.githubusercontent.com/35462038/195355429-05f24b48-470d-402e-af76-3079b5ab4a66.png) | ![after_no_spacing](https://user-images.githubusercontent.com/35462038/195355468-72f0105a-3555-4b49-96ea-2ef02396d96e.png) |

Reply:
Hmm.. from I can see is that the "Updated header with spacing" does align better with the current headers and spacing. However, I am not fully sure it if the spacing matched exactly to the current header and spacing as I was visually inspecting and drawing a horizontal line across some of the boundary lines across the 2 different screenshots.

One immediate issue that I want to point out is that we need to document in the Spacer that we are subtracting the 40dp and 16dp spacers by 12dp to account for the padding of this button. Otherwise, it would be too easy for someone to look at the figma designs and say this should be 40dp and revert these changes. I can understand from a coding perspective that it's unfortunate we can't be super explicit about just saying 40dp/16dp for these spacers to exactly match the figma padding specs. However, I understand that all of these buttons have some padding to them that makes them bigger than they seem due to the minimum touch surface and ripple effects.